program/tests/context.rs

```diff
@@ -881,6 +884,89 @@ impl Context {
+    pub async fn try_unstake(&mut self, amount: Lamports) -> transport::Result<()> {
+        // Where the new stake will live.
+        let solido = self.get_solido().await;
+        // solido.validators.entries[0].entry.get
+        let validator = solido
+            .validators
+            .get(&self.validator.as_ref().unwrap().vote_account)
+            .unwrap();
+
+        let (source_stake_account, _) = Validator::find_stake_account_address(
+            &id(),
+            &self.solido.pubkey(),
+            &validator.pubkey,
+            validator.entry.stake_seeds.stake_accounts_seed_begin,
+        );
+        let (destination_stake_account, _) = Validator::find_unstake_account_address(
+            &id(),
+            &self.solido.pubkey(),
+            &validator.pubkey,
+            validator.entry.unstake_seeds.stake_accounts_seed_begin,
+        );
+
+        let stake_account_before = self.get_stake_account_from_seed(&validator.pubkey, 0).await;
+
+        send_transaction(
+            &mut self.context,
+            &mut self.nonce,
+            &[
+                // authorize_burn_instruction,
+                instruction::unstake(
+                    &id(),
+                    &instruction::UnstakeAccountsMeta {
+                        lido: self.solido.pubkey(),
+                        validator_vote_account: self.validator.as_ref().unwrap().vote_account,
+                        source_stake_account,
+                        destination_stake_account,
+                        stake_authority: self.stake_authority,
+                        maintainer: self.maintainer.as_ref().unwrap().pubkey(),
+                    },
+                    amount,
+                ),
+            ],
+            vec![self.maintainer.as_ref().unwrap()],
+        )
+        .await?;
+
+        let stake_account_after = self.get_stake_account_from_seed(&validator.pubkey, 0).await;
+        assert_eq!(
+            (stake_account_before.balance.total() - stake_account_after.balance.total()).unwrap(),
+            amount
+        );
+        let unstake_account = self
+            .get_unstake_account_from_seed(&validator.pubkey, 0)
+            .await;
+
+        let mut deactivating_amount = amount;
+
+        let (unstake_addr, _) = Validator::find_unstake_account_address(
+            &id(),
+            &self.solido.pubkey(),
+            &validator.pubkey,
+            0,
+        );
+        let unstake_account_balance = self.get_sol_balance(unstake_addr).await;
+        // We can't unstake all the balance if we ask for it, the split will not
+        // deactivate the rent.
+        if unstake_account_balance == amount {
+            let rent = self.get_rent().await;
+            let stake_rent = rent.minimum_balance(std::mem::size_of::<StakeState>());
+            deactivating_amount = (amount - Lamports(stake_rent)).unwrap();
+        }
+        assert_eq!(unstake_account.balance.deactivating, deactivating_amount);
```

Comment:
This feels like it should be part of a test, not the test context. I would keep the execution part in here, but move the additional balance checks into the test.